crates/router/src/core/payment_methods/cards.rs

```diff
@@ -2204,6 +2196,62 @@ pub async fn call_to_locker_hs(
     Ok(stored_card)
 }
 
+#[instrument(skip_all)]
+pub async fn call_locker_api<T>(
+    state: &routes::SessionState,
+    request: Request,
+    flow_name: &str,
+    locker_choice: Option<api_enums::LockerChoice>,
+) -> errors::CustomResult<T, errors::VaultError>
+where
+    T: serde::de::DeserializeOwned,
+{
+    let locker = &state.conf.locker;
+    let jwekey = state.conf.jwekey.get_inner();
+    let response_type_name = type_name!(T);
+
+    let response = services::call_connector_api(state, request, flow_name)
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .change_context(errors::VaultError::ApiError)?;
```

Comment:
Why are we switching context to an internal server error and then immediately switching to `VaultError`?

Reply:
We can directly raise VaultError, will remove ApiErrorResponse